synadm/api.py

```diff
@@ -28,12 +28,39 @@
 https://matrix.org/docs/spec/#matrix-apis.
 """
 
-import requests
-from http.client import HTTPConnection
 import datetime
 import json
-import urllib.parse
 import re
+from typing import Optional, Union, Dict, List, Any
+import urllib.parse
+from http.client import HTTPConnection
+import traceback
+
+import requests
+from requests.exceptions import ConnectionError
+
+
+def log_fatal_exit(error, logger, message=None):
+    """Log a fatal error and exit synadm.
+
+    Args:
+        error: A Python exception.
+        logger: A Python logger, with info and fatal methods
+        message: Message to use instead of "synadm exited due to a fatal
+            error."
+    """
+    if message is None:
+        message = "synadm exited due to a fatal error."
+
+    # split by new lines that's already included
+    logger.info("".join(traceback.format_exception(error)))
```

Comment:
I had to read the docs to understand what's happening here and understand what you try to tell in the comment: https://docs.python.org/3/library/traceback.html#traceback.format_exception

Reply:
I also found this article about itL https://stackoverflow.com/questions/1378889/how-to-format-traceback-objects-in-python

Reply:
A minute after writing this, starting my work-day, it came to me "Wasn't this a logger.info(), of course I don't see it, it's in the log or with verbose on console. Sorry for the question. Got it :-)